packages/luigis-box/src/Model/Product/Filter/LuigisBoxFacetsToProductFilterOptionsMapper.php

```diff
@@ -0,0 +1,125 @@
+{
+    public const string FACET_AVAILABILITY = 'availability_rank_text';
+    public const string FACET_BRAND = 'brand';
+    public const string FACET_LABELS = 'labels';
+    public const string FACET_PRICE = 'price_amount';
+    public const array PRODUCT_FACET_NAMES = [
+        self::FACET_AVAILABILITY,
+        self::FACET_BRAND,
+        self::FACET_LABELS,
+        self::FACET_PRICE,
+    ];
+
+    /**
+     * @param \Shopsys\LuigisBoxBundle\Model\Brand\BrandRepository $brandRepository
+     * @param \Shopsys\LuigisBoxBundle\Model\Flag\FlagRepository $flagRepository
+     * @param \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterOptionsFactory $productFilterOptionsFactory
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterConfigFactory $productFilterConfigFactory
+     */
+    public function __construct(
+        protected readonly BrandRepository $brandRepository,
+        protected readonly FlagRepository $flagRepository,
+        protected readonly ProductFilterOptionsFactory $productFilterOptionsFactory,
+        protected readonly ProductFilterConfigFactory $productFilterConfigFactory,
+    ) {
+    }
+
+    /**
+     * @param array<int, array<string, mixed>> $luigisBoxFacets
+     * @param \Shopsys\FrameworkBundle\Model\Product\Filter\ProductFilterData $productFilterData
+     * @return \Shopsys\FrontendApiBundle\Model\Product\Filter\ProductFilterOptions
+     */
+    public function map(array $luigisBoxFacets, ProductFilterData $productFilterData): ProductFilterOptions
+    {
+        $brands = [];
+        $flags = [];
+        $priceRange = new PriceRange(Money::zero(), Money::zero());
+
+        $productFilterCountData = new ProductFilterCountData();
+
+        foreach ($luigisBoxFacets as $facetData) {
+            if ($facetData['name'] === self::FACET_AVAILABILITY) {
+                $productFilterCountData->countInStock = $facetData['values'][0]['hits_count'] ?? 0;
+            }
+
+            if ($facetData['name'] === self::FACET_BRAND) {
+                $brandCountsByName = $this->mapValuesToCountsByName($facetData['values']);
+                $brands = $this->brandRepository->getBrandsByNames(array_keys($brandCountsByName));
+
+                foreach ($brands as $brand) {
+                    $productFilterCountData->countByBrandId[$brand->getId()] = $brandCountsByName[$brand->getName()];
+                }
+            }
+
+            if ($facetData['name'] === self::FACET_LABELS) {
+                $flagCountsByName = $this->mapValuesToCountsByName($facetData['values']);
+                $flags = $this->flagRepository->getFlagsByNames(array_keys($flagCountsByName));
+
+                foreach ($flags as $flag) {
+                    $productFilterCountData->countByFlagId[$flag->getId()] = $flagCountsByName[$flag->getName()];
+                }
+            }
+
+            if ($facetData['name'] !== self::FACET_PRICE) {
+                continue;
```

Comment:
this early return actually worsens the readability IMHO but it is forced by our standards. right?

Reply:
Yes